cmd/relay-gossip/main.go

```diff
@@ -99,46 +100,157 @@ var runCmd = &cli.Command{
+		if len(groupConfs) >= 1 &&
+			len(hashesMap) == 0 {
+
+			for _, groupConf := range groupConfs {
+				err = cctx.Set(lib.GroupConfFlag.Name, groupConf)
+				if err != nil {
+					return fmt.Errorf("setting group-conf %q got: %w", groupConf, err)
+				}
+
+				err := boostrapGossipRelayNode(cctx, "")
+				if err != nil {
+					return err
+				}
 			}
-			chainHash = hex.EncodeToString(info.Hash())
 		}
 
-		cfg := &lp2p.GossipRelayConfig{
-			ChainHash:    chainHash,
-			PeerWith:     cctx.StringSlice(peerWithFlag.Name),
-			Addr:         cctx.String(listenFlag.Name),
-			DataDir:      cctx.String(storeFlag.Name),
-			IdentityPath: cctx.String(idFlag.Name),
-			Client:       c,
+		for chainHash, groupConf := range hashesMap {
+			if err := cctx.Set(lib.HashFlag.Name, chainHash); err != nil {
+				return fmt.Errorf("setting client hash: %w", err)
+			}
+
+			if groupConf != "" {
+				err = cctx.Set(lib.GroupConfFlag.Name, groupConf)
+				if err != nil {
+					return fmt.Errorf("setting group-conf for hash %q got: %w", chainHash, err)
+				}
+			}
+
+			if err := boostrapGossipRelayNode(cctx, chainHash); err != nil {
+				return err
+			}
 		}
-		if _, err := lp2p.NewGossipRelayNode(log.DefaultLogger(), cfg); err != nil {
-			return err
+
+		// Try and build a default gossip relay using other options that might have been set.
+		if len(hashesMap) == 0 &&
+			len(groupConfs) == 0 &&
+			!hashFlagSet {
+			if err := boostrapGossipRelayNode(cctx, ""); err != nil {
+				return err
+			}
 		}
-		<-chan int(nil)
-		return nil
+
+		// Wait until we are signaled to shutdown.
+		<-cctx.Context.Done()
+
+		return cctx.Context.Err()
 	},
 }
 
+func computeGroupConfs(cctx *cli.Context) []string {
+	if !cctx.IsSet(lib.GroupConfFlag.Name) {
+		return nil
+	}
+
+	groupConfs := cctx.String(lib.GroupConfFlag.Name)
+	return strings.Split(groupConfs, ",")
+}
+
+func boostrapGossipRelayNode(cctx *cli.Context, chainHash string) error {
+	c, err := lib.Create(cctx, cctx.IsSet(metricsFlag.Name))
+	if err != nil {
+		return fmt.Errorf("constructing client: %w", err)
+	}
+
+	chainInfo, err := c.Info(cctx.Context)
+	if err != nil {
+		return fmt.Errorf("cannot retrieve chain info: %w", err)
+	}
+
+	// Set the path to be desired 'storage path / beaconID'.
+	// This allows running multiple networks via the same beacon.
+	dataDir := path.Join(cctx.String(storeFlag.Name), chainInfo.ID)
+
+	cfg := &lp2p.GossipRelayConfig{
+		ChainHash:    chainHash,
+		BeaconID:     chainInfo.ID,
+		PeerWith:     cctx.StringSlice(peerWithFlag.Name),
+		Addr:         cctx.String(listenFlag.Name),
+		DataDir:      dataDir,
+		IdentityPath: cctx.String(idFlag.Name),
+		Client:       c,
+	}
+	_, err = lp2p.NewGossipRelayNode(log.DefaultLogger(), cfg)
+	if err != nil {
+		err = fmt.Errorf("could not initialize a new gossip relay node %w", err)
+	}
+	return err
+}
+
+func computeHashesMap(hashMaps, groupConfs []string) (map[string]string, error) {
+	hashesMap := make(map[string]string)
```

Comment:
the naming of `hashMaps` and `hashesMaps` made me do a double take

Reply:
Fixed